src/orange/OrangeTypes.hh

```diff
@@ -101,6 +101,18 @@ enum class Sense : bool
     outside,  //!< Expression is greater than zero
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Transformations to apply to senses when using lazy sense evaluation.
+ */
+enum class SenseMod : unsigned char
+{
+    normal = 0,
+    flipped = 1 << 0,
+    cached = 1 << 1,
```

Comment:
I was also thinking of combining the two, however, many classes rely on the `OrangeTrackView` API returning a `Span<Sense>`, so this change would require updating many users of that interface.  Maybe we could make an on-the-fly conversion of `Span<FlaggedSense>` to `Span<Sense>`

Reply:
Is it legal to reinterpret cast `FlaggedSense*` to `Sense*`?  Maybe we could have a function that checks that the reinterpreted values are equivalent and performs the span cast? And I see now that `Sense` has an underlying type of `bool`, but since that still has sizeof(1) should it be ok?

Reply:
I don't think `reinterpret_cast` would work here, any non-zero value, i.e., having any flag set would be interpreted as true. We'd have to check the sense bit.

Reply:
maybe we could have something like: 
```c++
struct Sense
{
bitset<2> sense_;
operator bool() const
{
return sense_[SenseFlags::sense];
}
};

enum SenseFlags : unsigned char
{
    cached = 0x1,  //!< Sense is cached
    sense = 0x2,  //!< Sense
};
```

instead of the Sense being a simple enum.